src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -375,9 +387,17 @@ private static int BioRead(SafeBioHandle bio, IntPtr buffer, int count)
         }
 
         //TODO (Issue #3362) should we check Bio should retry?
-        private static int BioWrite(SafeBioHandle bio, IntPtr buffer, int count)
+        private static int BioWrite(SafeBioHandle bio, byte[] buffer, int offset, int count)
         {
-            int bytes = libssl.BIO_write(bio, buffer, count);
+            int bytes;
```

Comment:
Ditto on the asserts comment (there are probably some other places, as well... this can all be done as part of a separate change, too)